slick-testkit/src/main/scala/com/typesafe/slick/testkit/util/StandardTestDBs.scala

```diff
@@ -292,6 +295,9 @@ object DerbyDB {
 abstract class HsqlDB(confName: String) extends InternalJdbcTestDB(confName) {
   val profile = HsqldbProfile
   val jdbcDriver = "org.hsqldb.jdbcDriver"
+  // Hsqldb has valid "select for update" syntax, but in testing, it either takes a whole table lock or no exclusive
+  // lock at all, so exclude from ForUpdate testing
+  override def capabilities = super.capabilities - TestDB.capabilities.selectForUpdateRowLocking
```

Comment:
Oh, wait, this one is in TestDB.capabilities. So let me rephrase my last comment: Why is it separate from `rowLocking`? If HsqlDB doesn't properly lock in these cases, I think it shouldn't advertise `rowLocking` in the first place.